src/main/kotlin/snyk/iac/annotator/IacBaseAnnotator.kt

```diff
@@ -17,7 +17,16 @@ private val LOG = logger<IacBaseAnnotator>()
 
 abstract class IacBaseAnnotator : ExternalAnnotator<PsiFile, List<IacIssue>>() {
 
-    abstract fun textRange(psiFile: PsiFile, iacIssue: IacIssue): TextRange
+    override fun collectInformation(psiFile: PsiFile): PsiFile = psiFile
+
+    override fun doAnnotate(psiFile: PsiFile?): List<IacIssue> {
+        LOG.debug("Calling doAnnotate on ${psiFile?.name}")
+
+        val iacResult = psiFile?.project?.service<SnykToolWindowPanel>()?.currentIacResult ?: return emptyList()
+        ProgressManager.checkCanceled()
+
+        return getAnnotationForFile(psiFile, iacResult)
+    }
```

Comment:
It's javadoc says " implementations should either avoid accessing indices and PSI or perform needed checks and locks themselves."
PS We'll actually avoid that restriction by using ordinary Annotator here.